src/shell/shell_batch.cpp

```diff
@@ -61,35 +61,41 @@ bool BatchFile::ReadLine(char * line) {
 	}
 	DOS_SeekFile(file_handle,&(this->location),DOS_SEEK_SET);
 
-	Bit8u c=0;Bit16u n=1;
-	char temp[CMD_MAXLINE];
+	uint16_t bytes_read = 1;
+	uint8_t data = 0;
+	char val = 0;
+	char temp[CMD_MAXLINE] = "";
 emptyline:
 	char * cmd_write=temp;
 	do {
-		n=1;
-		DOS_ReadFile(file_handle,&c,&n);
-		if (n>0) {
+		bytes_read = 1;
+		DOS_ReadFile(file_handle, &data, &bytes_read);
+		assert(data <= CHAR_MAX);
```

Comment:
Not sure if this assert is correct… I think different codepages utilize codepoints beyond 127 for letters.

Reply:
And even in english… .bat files often print non-letter characters like frames, textual "windows", etc - e.g. many GOG game releases do it.

Reply:
It's implicitly cast to a char when it's appended to `*cmd_write++ = val;` a bit further down.

Reply:
> bat files often print non-letter characters like frames, textual "windows"

I guess this should allow the wrap-around then, and then presumably there's some dos code that casts back to unsigned and the (negative char values) get unwrapped back up to > 127, where they're interpreted and printed.

Reply:
Replaced with `assert(data <= CHAR_MAX)`, which will let those high ASCII values wrap into the negative char range (which I suppose they will be dormant, unless passed into another DOS API and they unwrap back into the positive?).